### ISSUE TITLE
Bug: train_test_split() called without random_state — results are non-reproducible

### ISSUE THREAD

**Issue by Daksh7785** [SdSarthak/AegisAI#515]
## Bug Report

**Description**
Multiple projects call `train_test_split()` without setting `random_state`, meaning the
train/test split changes on every run. This produces different model accuracy each run,
making results impossible to reproduce or compare — especially bad in a learning repository
where students expect consistent output.

## Affected Files

| File | Line |
|------|------|
| `Bitcoin Price Prediction Web App/app.py` | 50 |
| `Chicken_Disease_Classification/research/mlflow_exp/example.py` | 47 |
| `Social Media Fake Accounts Detection/jserver/model.py` | 76 |
| `Ensemble Methds in ML/voting.py`, `blending.py`, `stacking.py` | various |

## Code (Buggy)
```python
xtrain, xtest, ytrain, ytest = train_test_split(X, Y, test_size=0.2)
```

## Fix
```python
xtrain, xtest, ytrain, ytest = train_test_split(X, Y, test_size=0.2, random_state=42)
```

## Severity
Medium — non-reproducible experiments; misleads beginners learning ML

**Comment by snowlindhanusha** [comment]
Hi, I am a GSSoC’26 contributor. I would like to work on this issue. Please assign it to me.

**Comment by akhila-sraj** [comment]
Hi @Daksh7785 , I am a GSSoC 2026 contributor and would like to work on this issue.

I have experience with cybersecurity concepts and secure application development, and I’d love to contribute. Please assign this to me.

**Comment by TejeeshRagnarok** [comment]
Hi @Daksh7785 and maintainers, I’d like to work on this issue as part of GSSoC'26.

I have previous backend/AI contribution experience in AegisAI (#400 merged PR), and I’d like to improve the ML pipeline reproducibility by fixing the missing `random_state` usage in `train_test_split()` calls and validating deterministic behavior across runs.

**Comment by madsysharma** [comment]
Hi @SdSarthak , I would like to work on this as part of GSSoC 2026.

**Comment by YerraguntaAjayKumar** [comment]
Hi @SdSarthak and @Daksh7785, I am a GSSoC '26 contributor and would love to resolve this issue to ensure ML pipeline reproducibility across the repository!

Here is my targeted implementation plan to fix the missing seeds:
1. **Deterministic Splitting:** I will audit and update the explicit `train_test_split()` calls across all 4 affected modules listed (`Bitcoin Price Prediction Web App`, `Chicken_Disease_Classification`, `Social Media Fake Accounts Detection`, and `Ensemble Methds in ML`) to append a standard `random_state=42` parameter.
2. **Pipeline Verification:** Ensure that running these files sequentially yields consistent, predictable model validation accuracy metrics across environment setups without shifting dataset distribution boundaries.
3. **Clean Code Isolation:** I'll keep the changes strictly isolated to the exact lines of code where the splits occur, ensuring absolutely zero linting, formatting, or lockfile churn.

Please assign this issue to me so I can get a clean, reliable PR submitted for review!

**Comment by manvibuilds** [comment]
Hi @Daksh7785 , I am a GSSoC 2026 contributor and would like to work on this issue.
I can add a random_state to every train_test_split() call and check if the results are reproducible.
Can I be assigned to this task?

**Comment by agsaru** [comment]
@SdSarthak Can You assign me this issue under GSSoC'26?

**Comment by manvibuilds** [comment]
Hey @SdSarthak  I have searched whole code base and the only file having train_test_split is backend/app/modules/guard/training/data/split. py with random_state=42 on both functions. The files listed in the issue( Bitcoin Price Prediction, Chicken disease...) on this repo doesn't seem to exist. Could you please clarify it?

**Comment by agsaru** [comment]
@Daksh7785  What is this non sense issue from . The files or folders do not even exists.

@SdSarthak Please close this type non sense issues.

**Comment by Daksh7785** [comment]
> [@Daksh7785](https://github.com/Daksh7785) What is this non sense issue from . The files or folders do not even exists.
> 
> [@SdSarthak](https://github.com/SdSarthak) Please close this type non sense issues.

When someone not able to solve then always says that issue was not genuine or nonsense

**Comment by agsaru** [comment]
@Daksh7785 ok then show me the file location in the repository. Adding just a random_state is not a big task ok.

**Comment by Daksh7785** [comment]
> [@Daksh7785](https://github.com/Daksh7785) ok then show me the file location in the repository. Adding just a random_state is not a big task ok.
If don't have abilities to solve issues then avoid it 
Don't waste the other person time
